service/src/IOUringImp.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef IOURINGIMP_HPP_INCLUDE
+#define IOURINGIMP_HPP_INCLUDE
+
+#define _POSIX_C_SOURCE 200809L
```

Comment:
Looks like maybe this needs to be _DEFAULT_SOURCE instead, for `loff_t`.

Reply:
I think we define _POSIX_C_SOURCE in configure.ac as defined for all compilation units.  We may have to be careful about how these settings interact.

Reply:
From https://man7.org/linux/man-pages/man7/feature_test_macros.7.html, I think `_DEFAULT_SOURCE` is similar to `-D_BSD_SOURCE -D_SVID_SOURCE -D_POSIX_C_SOURCE=200809`. Should I switch the configure option to use that instead?

Reply:
We define it in Makefile.am:

https://github.com/geopm/geopm/blob/dev/service/Makefile.am#L7-L9

I wonder if it would be best to just put

```
#define _BSD_SOURCE
#define _SVID_SOURCE
```
in the IOUringImp.hpp

and rely on the _POSIX_C_SOURCE that gets propagated through AM_CPPFLAGS